migrations/V147__System_Intake_Disallow_Invalid_state.sql

```diff
@@ -0,0 +1,7 @@
+ALTER TABLE system_intakes
+ADD CONSTRAINT system_intake_decision_step_requires_a_decision
+    CHECK ((step <> 'DECISION_AND_NEXT_STEPS'::system_intake_step) OR (step = 'DECISION_AND_NEXT_STEPS'::system_intake_step AND decision_state <> 'NO_DECISION'::system_intake_decision_state));
```

Comment:
I _think_ this can technically be simplified to `CHECK ((step <> 'DECISION_AND_NEXT_STEPS'::system_intake_step) OR decision_state <> 'NO_DECISION'::system_intake_decision_state))`, but this is a bit more clear about the intent, so I think it's fine.